app/controllers/locations_controller.rb

```diff
@@ -105,11 +104,10 @@ def locations_by_user
 
   # Display list of locations that a given user is editor on.
   def locations_by_editor
-    user = if params[:id]
-             find_or_goto_index(User, params[:by_editor].to_s)
-           else
-             @user
-           end
+    user = find_obj_or_goto_index(
+      model: User, obj_id: params[:by_editor].to_s,
+      index_path: locations_path
+    )
```

Comment:
Changes functionality when by_user is not an existing user.
See [Comment about LocationsController](https://github.com/MushroomObserver/mushroom-observer/pull/1324/files#r1095161355).